.circleci/config.yml

```diff
@@ -0,0 +1,137 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  environment:
+    LEIN_ROOT: "true"
+    JVM_OPTS: -Xmx3200m
+
+jobs:
+  checkout_code:
+    docker:
+      - image: circleci/clojure:lein-2.9.0
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-dependencies-{{ checksum "project.clj" }}
+            - v2-dependencies-
+      - run:
+          name: Downloading deps
+          command: |
+            lein deps
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: v2-dependencies-{{ checksum "project.clj" }}
+      - persist_to_workspace:
+          root: /home/circleci
+          paths:
+            - .m2
+            - repo
+
+  lint_code:
+    docker:
+      - image: circleci/clojure:lein-2.9.0
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install make
+          command: |
+            sudo apt-get install make
+      - run:
+          name: Running Eastwood
+          command: |
+            make eastwood
+      - run:
+          name: Running cljfmt
+          command: |
+            make cljfmt
+
+  test_code:
+    description: Run tests
+    parameters:
+      jdk_version:
+        description: Version of JDK to test against
+        type: string
+        default: ""
+      clojure_version:
+        description: Version of Clojure to test against
+        type: string
+    docker:
+      - image: circleci/clojure:<< parameters.jdk_version >>lein-2.9.0
+    <<: *defaults
+    environment:
+      VERSION: << parameters.clojure_version >>
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install make
+          command: |
+            sudo apt-get install make
+      - run:
+          name: Running tests
+          command: make test
+
+
+workflows:
+  version: 2.1
+  build_and_test:
+    jobs:
+      - checkout_code
+      - lint_code:
+          name: Code Linting
+          requires:
+            - checkout_code
+      - test_code:
+          name: java8, clj1.8 tests
```

Comment:
What would `name` default to if we didn't set it?

Reply:
It would come out as `test_code-1`, `test_code-2`, etc.

Reply:
:+1: